app/extras/queries/users_to_mention_query.rb

```diff
@@ -0,0 +1,5 @@
+class Queries::UsersToMentionQuery
+  def self.for(model)
+    model.mentioned_group_members - model.group.users.mentioned_in(model)
```

Comment:
This is a bit different, but much easier to track. It goes 'The users to mention are the group members mentioned in this body, sans the group members who have been mentioned in this model before (<-- users who have a 'user_mentioned' event associated with this model)'

I _think_ this is way better than how we were doing it before, which was to look at the previous iteration and determine a diff based on the two sets of mentions.

The subtle change in behaviour here is we lose is the case 'I mention @bob. He is notified. I remove @bob and mention @alice. She is notified. I remove @alice and mention @bob again. He is notified again.'

^^ I think not having that is perfectly acceptable behaviour, and quite possibly better.